src/Core_Command.php

```diff
@@ -586,6 +589,10 @@ function wp_new_blog_notification() {
 			'admin_password' => '',
 		];
 
+		if ( \WP_CLI\Utils\wp_version_compare( '4.0', '>=' ) ) {
+			$defaults['locale'] = '';
```

Comment:
The code should check if `--locale` is set here and display a warning along the lines of:
```
Warning: The flag --locale=de_DE is being ignored as it requires WordPress 4.0+.
```